spec.html

```diff
@@ -155,4 +155,209 @@ <h1>Array.prototype.with ( _index_, _value_ )</h1>
             </emu-clause>
         </emu-clause>
     </emu-clause>
+
+    <emu-clause id="sec-%typedarray%-intrinsic-object">
+      <h1>The %TypedArray% Intrinsic Object</h1>
+
+        <emu-clause id="sec-properties-of-the-%typedarrayprototype%-object">
+            <h1>Properties of the %TypedArray% Prototype Object</h1>
+
+            <emu-clause id="sec-%typedarray%.prototype.popped">
+                <h1>%TypedArray%.prototype.popped ( )</h1>
+
+                <p>When the *popped* method is called, the following steps are taken:</p>
+
+                <emu-alg>
+                    1. Let _O_ be the *this* value.
+                    1. Perform ? ValidateTypedArray(_O_).
+                    1. Let _len_ be _O_.[[ArrayLength]].
+                    1. Let _newLen_ be max(_len_ - 1, 0).
+                    1. Let _A_ be ? TypedArraySpeciesCreate(_O_, &laquo; 𝔽(_newLen_) &raquo;).
+                    1. Let _k_ be 0.
+                    1. Repeat, while _k_ &lt; _newLen_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _kValue_ be ! Get(_O_, _Pk_).
+                        1. Perform ? Set(_A_, _Pk_, _kValue_, *true*).
+                        1. Set _k_ to _k_ + 1.
+                    1. Return _A_.

```

Comment:
seems like these lines could be an abstract operation?

Reply:
abstract operation added.